### PR TITLE
NL: Implement disambiguation places and related places/vars for real

### PR DESCRIPTION
**Pull request by pradh**
Three new types of suggestions. Place suggestions are shown only if the current query includes a place, and likewise for SVs.

* **Disambiguation places**
   - These are places in the 2nd index onwards in RecognizePlaces API result
   - For each such place, we display the parent place
      - For California - where alternatives are cities - "City in Washington" is displayed
      - For Franklin County - where alternatives are all counties - "Pennsylvania" is displayed

* **Related places**
   - These are places in the cohort-set as the main place.  Uses the same logic as [getSimilarPlaces in Mixer](https://github.com/datacommonsorg/mixer/blob/af42174d937bb6ee089ca6e000c5d9aa9f047dcd/internal/server/v1/page/place.go#L512-L513)
   - For these, we just display the name

* **Related variables**
    - These are SVs returned from the embeddings (that is important!), past a certain chart distance on the page (6 currently).  They may include SVs that aren't in the charts too.
    - For these, we just display the name of the SV.

TODO: Ensure the places are added to existence check.

### CODE REVIEW

**Review by jehangiramjad — APPROVED** [review]
Thanks! Generally look good. A few comments/suggestions/clarifications

File: server/lib/nl/detection/place.py [hunk 1]
```diff
@@ -271,9 +289,34 @@ def _set_query_detection_debug_logs(d: PlaceDetection,
...
+    query_detection_debug_logs["similar_places"] = \
+      '; '.join([p.name for p in d.similar_to_main_place])
   if not query_detection_debug_logs["place_dcid_inference"]:
     query_detection_debug_logs[
         "place_dcid_inference"] = "Place DCID Inference did not trigger (no place strings found)."
   if not query_detection_debug_logs["place_resolution"]:
     query_detection_debug_logs[
         "place_resolution"] = "Place resolution did not trigger (no place dcids found)."
+
+
```
[the remark this answers is not in the document]
Comment (pradh) [reply server/lib/nl/detection/place.py:305]: Sorry added.

File: server/lib/nl/detection/place.py [hunk 2]
```diff
@@ -271,9 +289,34 @@ def _set_query_detection_debug_logs(d: PlaceDetection,
...
+def _get_places_for_listpair(similars, identicals):
+  if not similars and not identicals:
+    return [], []
+
+  merged = list(set(similars + identicals))
+
+  dummy = {}
```
[the remark this answers is not in the document]
Comment (pradh) [reply server/lib/nl/detection/place.py]: oh, dinno, s/dummy/tmp/

File: server/lib/nl/detection/place_utils.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  res = fetch.raw_property_values([cohort], 'member')
+  members = []
+  for m in res.get(cohort, []):
+    if 'dcid' not in m:
+      continue
+    members.append(m['dcid'])
+  if not members:
+    return []
+
+  # Randomize order.
```
[the remark this answers is not in the document]
Comment (pradh) [reply server/lib/nl/detection/place_utils.py:49]: I think to make it interesting? :)  This is what the Mixer impl does when showing related stuff on our place pages.

**Review by pradh — COMMENTED** [review]
Thanks for the review Jehangir!

**Pull request merged**